### PR TITLE
apiVersion shouldn't be hardcoded

### PR DESCRIPTION
**Pull request by DStorck**
**What this PR does / why we need it**:
Cronjobs have different api versions for kubernetes versions < 1.8, so it shouldn't be hardcoded into the chart. This PR : 
- adds info to readme clarifying api version of cronjobs
- adds cronjob.apiVersion to _helpers.tpl file to check k8s version before deploy

### CODE REVIEW

File: incubator/elasticsearch-curator/templates/_helpers.tpl [hunk 1]
```diff
@@ -1,5 +1,16 @@
 {{/* vim: set filetype=mustache: */}}
 
+{{/*
+Return the appropriate apiVersion for cronjob APIs.
+*/}}
+{{- define "cronjob.apiVersion" -}}
+{{- if semverCompare "< 1.8" .Capabilities.KubeVersion.GitVersion -}}
+{{- print "batch/v2alpha1" }}
+{{- else if semverCompare ">=1.8" .Capabilities.KubeVersion.GitVersion -}}
```
Comment (mattfarina) [inline incubator/elasticsearch-curator/templates/_helpers.tpl]: Can you add a `-0` to the end of the comparison (e.g. `>=1.8-0`). This will cause the semver comparison to evaluate pre-releases. Unfortunately, some of the distros use pre-releases to specify their version which can cause comparison breakages. Pre-releases, as defined by semver, are prior to the release, lower in order, and may not satisfy the releases API which is why they are ignored by default.

You can find another PR that does this as an example at #4605

**Review by mattfarina — COMMENTED** [review]
/ok-to-test

Thanks for the PR. This almost looks good but we found an edge case where this could break in some environments.

File: incubator/elasticsearch-curator/templates/_helpers.tpl (hunk 1 above)
Comment (DStorck) [reply incubator/elasticsearch-curator/templates/_helpers.tpl]: thanks, done

**Comment by gianrubio** [comment]
> Cronjobs have different api versions for kubernetes versions < 1.8, so it shouldn't be hardcoded into the chart. This PR :

We had this discussion [before this PR was merged](https://github.com/kubernetes/charts/pull/4515#discussion_r178322711) so that's why the cronjob api version was removed. Other than that PR looks good

/ok-to-test
/lgtm

**Review by gianrubio — APPROVED** [review]
